### ISSUE TITLE
Large Library Causes Lockup?

### ISSUE THREAD

**Issue by Keith-Cancel** [DeaDBeeF-for-Windows/deadbeef#16]
### Steps to reproduce the problem
I try loading a large music library, hundreds of gigabytes of music.
### What's going on? Describe the problem in as much detail as possible.
I suspect it's the windows event loop is not being updated enough? Perhaps directory scanning/search should be an other thread. If it is not already.

I probably should let it sit for quite some time to see if it finishes. If that is the cases aka it's not calling TranslateMessage() and DispatchMessageW() often enough to keep the gui responsive for windows. This is just a guess though. I am not that familiar with this projects internals.

### Information about the software:

Deadbeef version: version 1.8.3
OS: Windows 10

![Lockup](https://user-images.githubusercontent.com/8355441/78647976-d4e87900-78aa-11ea-94f1-f8c9f9106ff1.png)

**Comment by Keith-Cancel** [comment]
Well interesting, it's not what I initially thought it was. One the directories had a cue sheet that referred to it's self. I went to adding music by groups at a time. Easy for me edit for to avoid the lock up, but perhaps it might be wise to add some circular reference detection. 

```
REM DATE 2008.10.13, M3-22
REM COMMENT DYVK-0001, 1st Single
PERFORMER "AYUTRICA"
TITLE "二つの影 -Zwei Schatten-"
FILE "二つの影 -Zwei Schatten-.cue" WAVE
  TRACK 01 AUDIO
    TITLE "二つの影 -- Zwei Schatten"
    PERFORMER "AYUTRICA"
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    TITLE "闇の回廊 -- Der Kreuzgang der Finsternis"
    PERFORMER "AYUTRICA"
    INDEX 01 04:35:23
```

**Comment by Keith-Cancel** [comment]
Okay I found an other thing that makes it it crash/lock up. I tend to store music videos or PVs with their associated albums. I found an mp4 of mine causes a lockup/crash when scanning a directory. Not sure why this mp4 specifically. It seems fine with other ones from a quick test. Also is there any way to exclude files from the scanning when adding a folder via regex/wildcard?

**Comment by kuba160** [comment]
So 2 things crashing the scanning:
- Cue files referring to themselves
- mp4 music videos (mp4s with video/audio?)

I will check it. I doubt there is any way to exclude files from scanning. Thank you for the information.

**Comment by kuba160** [comment]
It would be helpful if you would tell what codecs your mp4 files contain (the ones that hang). You can check it for example in vlc by rightclicking and pressing codec information:
![image](https://user-images.githubusercontent.com/6359901/78658600-a2e70f00-78ca-11ea-819d-516a61566c62.png)

**Comment by kuba160** [comment]
It would be helpful if you would tell what codecs your mp4 files contain (the ones that hang). You can check it for example in vlc by rightclicking and pressing codec information:
![image](https://user-images.githubusercontent.com/6359901/78658600-a2e70f00-78ca-11ea-819d-516a61566c62.png)

**Comment by Keith-Cancel** [comment]
Sure thing this is what VLC shows for the file.

![codec](https://user-images.githubusercontent.com/8355441/78660762-2e59a380-78bd-11ea-91d9-7e3b62d08614.png)

**Comment by kuba160** [comment]
- [x] CUE issue is persisent across all platforms, reported to DeaDBeeF project.
- [x] MP4 issue is located in ALAC plugin, reported to DeaDBeeF project as well. (fixed)

**Comment by kuba160** [comment]
Fixed in fb7048c717a18155576475abe0a783c1ac5c7a4b and a6ac452135814c3ff64206440433a085a10bb36c.
